### PR TITLE
Add new PGP subkey needed for nightly 2021-05-26-21-17-56 and later

### LINKED ISSUE #257: What changed regarding GPG keys for releases?
I maintain https://github.com/colemickens/flake-firefox-nightly which provides nightly builds to nix flakes users.

However, something broke regarding Firefox releases. I'm on vacation and can't dig into this, but would like to not leave my users on a stuck version of Nightly for 5+ days.

Can anyone advise if something changed regarding how Firefox publishes releases?

```
error: builder for '/nix/store/xff2q0yjxxzmgbrnc0bjnyhkmfrcsp5q-firefox-90.0a1.en-US.linux-x86_64.tar.bz2.drv' failed with exit code 2;
       last 10 log lines:
       > gpg: key 61B7B526D98F0353: 26 signatures not checked due to missing keys
       > gpg: /build/tmp.jAJ9Gzdf8M/.gnupg/trustdb.gpg: trustdb created
       > gpg: key 61B7B526D98F0353: public key "Mozilla Software Releases <release@mozilla.com>" imported
       > gpg: Total number processed: 1
       > gpg:               imported: 1
       > gpg: no ultimately trusted keys found
       > gpgv: can't allocate lock for '/build/tmp.jAJ9Gzdf8M/.gnupg/pubring.kbx'
       > gpgv: Signature made Thu May 27 11:09:26 2021 UTC
       > gpgv:                using RSA key 4360FE2109C49763186F8E21EBE41E90F6F12F6D
       > gpgv: Can't check signature: No public key
       For full logs, run 'nix log /nix/store/xff2q0yjxxzmgbrnc0bjnyhkmfrcsp5q-firefox-90.0a1.en-US.linux-x86_64.tar.bz2.drv'.
error: build of '/nix/store/rb0g83kg404ngzpmbjjf7fyf26shnyl2-firefox-bin-90.0a1.drv', '/nix/store/x79gjl4qhrw7a54by099whbh0azccay5-firefox-release-bin-unwrapped-90.0a1.drv', '/nix/store/xff2q0yjxxzmgbrnc0bjnyhkmfrcsp5q-firefox-90.0a1.en-US.linux-x86_64.tar.bz2.drv' failed
nix-build-uncached: nix build failed: exit status 1
```

### PR DESCRIPTION
**Pull request by andersk**
Fixes #257.

### CODE REVIEW

**Comment by colemickens** [comment]
Is the confirmation in the Bugzilla thread enough to merge this?

**Comment by nbp** [comment]
@colemickens I will ask confirmation from release managers and if not from the security team

**Comment by nbp** [comment]
I've got the confirmation from a release manager.
However, the testing manually the downloaded file mentioned in this patch and computing the sha256 of it yields a different value.

**Comment by nbp** [comment]
Apparently `builtins.fetchurl` does check the sha but computes a different one in a different base than the hexadecimal output of `sha256sum`.

File: firefox-overlay.nix [hunk 1]
```diff
@@ -157,7 +154,7 @@ in
 {
   lib = super.lib // {
     firefoxOverlay = {
-      inherit firefoxVersion versionInfo firefox_versions;
+      inherit pgpKey firefoxVersion versionInfo firefox_versions;
```
Comment (nbp) [inline firefox-overlay.nix:157]: Thanks for adding this!

**Review by nbp — APPROVED** [review]

**Comment by andersk** [comment]
FYI, Nix also accepts the hash in base 16 (`bc463b3ee9603971042a5088b28a632d3e525b97d5c6f7ff528a6c521066a3fe`) and base 64 (`vEY7PulgOXEEKlCIsopjLT5SW5fVxvf/UopsUhBmo/4=`), in case that makes it easier to update in the future.

**Comment by nbp** [comment]
Thanks for the link! In the future I suggest to use either the base 16 (`sha256sum`) or the [SRI representation](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity), instead of the non-standard base32.

While the base32 is useful for directory storage, the checksum written in this file is meant to maximize our confidence of integrity.

**Pull request merged**